### PR TITLE
NRT

### PR DESCRIPTION
**Pull request by stefannikolei**
Contains work for https://github.com/SixLabors/ImageSharp/pull/2236

### CODE REVIEW

**Comment by stefannikolei** [comment]
@JimBobSquarePants I think this needs to be merged first. After this changes are in the repo my PR with nullable changes should hopefully compile. (IT works on my machine ;) )

File: src/SharedInfrastructure/Guard.cs [hunk 1]
```diff
@@ -20,16 +21,9 @@ internal static partial class Guard
...
-        where TValue : class
-    {
-        if (value is not null)
-        {
-            return;
-        }
-
-        ThrowHelper.ThrowArgumentNullExceptionForNotNull(parameterName);
-    }
+    public static void NotNull<TValue>([NotNull]TValue? value, string parameterName)
```
Comment (JimBobSquarePants) [inline src/SharedInfrastructure/Guard.cs:24]: We should delete `ThrowHelper.ThrowArgumentNullExceptionForNotNull(...)` now that it is not in use.

**Comment by stefannikolei** [comment]
@JimBobSquarePants This could be the next step for NRT. 
When this annotations make it to the main Repo, some NRT Errors will disappear.

But because of the Conditional Attribute the ArgumentNullException will not be thrown in Release builds. Is this intended?

**Comment by JimBobSquarePants** [comment]
I’m not sure I understand your question? We have to guard classes. `Guard` and `DebugGuard` the latter designed for debugging only.

**Comment by stefannikolei** [comment]
The DebugGuard will only execute while Debugging. So in release mode the method will not be executed and no ArgumentNullException will be thrown. Perhaps a NullReferenceException when the value is used which was passed to the DebugGuard

**Comment by JimBobSquarePants** [comment]
That’s by design. We don’t want DebugGuard running in release mode. 

It’s likely with the proper nullable fixes we won’t even require it.

**Review by JimBobSquarePants — APPROVED** [review]

**Pull request merged**